packages/rocketchat-lib/server/lib/interceptDirectReplyEmails.js

```diff
@@ -0,0 +1,124 @@
+const IMAP = require('imap');
+
+class IMAPIntercepter {
+	constructor() {
+		this.imap = new IMAP({
+			user: RocketChat.settings.get('Direct_Reply_Username'),
+			password: RocketChat.settings.get('Direct_Reply_Password'),
+			host: RocketChat.settings.get('Direct_Reply_Host'),
+			port: RocketChat.settings.get('Direct_Reply_Port'),
+			debug: RocketChat.settings.get('Direct_Reply_Debug') ? console.log : false,
+			tls: !RocketChat.settings.get('Direct_Reply_IgnoreTLS'),
+			connTimeout: 30000,
+			keepalive: true
+		});
+	}
+
+	openInbox(Imap, cb) {
+		Imap.openBox('INBOX', false, cb);
+	}
+
+	start() {
+		const self = this;
+		const Imap = this.imap;
+
+		Imap.connect();
+		// On successfully connected.
+		Imap.once('ready', Meteor.bindEnvironment(function() {
+			if (Imap.state !== 'disconnected') {
+				self.openInbox(Imap, Meteor.bindEnvironment(function(err) {
+					if (err) {
+						throw err;
+					}
+					// fetch new emails & wait [IDLE]
+					self.getEmails(Imap);
+
+					// If new message arrived, fetch them
+					Imap.on('mail', Meteor.bindEnvironment(function() {
+						self.getEmails(Imap);
+					}));
+				}));
+			} else {
+				console.log('IMAP didnot connected.');
+				Imap.end();
+			}
+		}));
+
+		Imap.once('error', function(err) {
+			console.log(err);
+			throw err;
+		});
+	}
+
+	isActive() {
+		const Imap = this.imap;
+		if (Imap.state === 'disconnected') {
+			return false;
+		}
+
+		return true;
+	}
+
+	stop(callback) {
+		const Imap = this.imap;
+		Imap.end();
+		Imap.once('end', function() {
```

Comment:
If you use the bindEnviroment here it's not necessary to use when calling the stop.

Try to use arrow functions instead of functions when there is no need to access the scope (this) of that function
`Imap.once('end', function() {` -> `Imap.once('end', () => {`

Use default parameters, like here you can turn the callback optional 
```javascript
stop(callback = new Function) {
```